src/data/bucket/fill_extrusion_bucket.js

```diff
@@ -173,10 +166,34 @@ class FillExtrusionBucket implements Bucket {
                             segment.primitiveLength += 2;
                         }
                     }
+                }
+            }
+
+            if (segment.vertexLength + numVertices > MAX_VERTEX_ARRAY_LENGTH) {
+                segment = this.segments.prepareSegment(4, this.layoutVertexArray, this.indexArray);
+            }
+
+            const flattened = [];
+            const holeIndices = [];
+            const indices = [];
+
+            for (const ring of polygon) {
+                if (ring.length === 0) {
+                    continue;
+                }
+
+                if (ring !== polygon[0]) {
+                    holeIndices.push(flattened.length / 2);
+                }
+
+                for (let i = 0; i < ring.length; i++) {
+                    const p = ring[i];
+
+                    addVertex(this.layoutVertexArray, p.x, p.y, 0, 0, 1, 1, 0);
+                    indices.push(segment.vertexLength++);
```

Comment:
You can avoid accumulating the `indices` array by caching `const startingIndex = segment.vertexLength` prior to the `ring of polygon` loop, and then calculating indexes like `startingIndex + triangleIndices[j]` after earcut. (See `FillBucket` for example.)